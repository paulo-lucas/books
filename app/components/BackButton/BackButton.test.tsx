import 'react-native';
import React from 'react';
import { render, screen } from '../../../tests/test-utils';
import { BackButton } from './index';

const mockGoBack = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualRNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...actualRNavigation,
    useNavigation: () => ({
      navigation: {
        goBack: mockGoBack,
      },
    }),
  };
});

describe('back button', () => {
  it('should render correctly', async () => {
    render(<BackButton />);
    expect(screen.toJSON()).toMatchSnapshot('book');
  });
});
