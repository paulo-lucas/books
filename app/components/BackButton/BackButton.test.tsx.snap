// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`back button should render correctly: book 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      android_ripple={
        Object {
          "borderless": false,
          "color": "rgba(0, 0, 0, 0.3)",
          "radius": -5,
        }
      }
      disabled={false}
      replaceTheme={[Function]}
      style={
        Object {
          "marginRight": 20,
          "padding": 10,
        }
      }
      testID="RNE__ICON__CONTAINER_ACTION"
      updateTheme={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#2089dc",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          î˜œ
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
